graphql/src/main/java/com/linecorp/armeria/server/graphql/DefaultGraphqlErrorHandler.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.graphql;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import javax.annotation.Nonnull;
+
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.armeria.internal.server.graphql.protocol.GraphqlUtil;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+import graphql.ExecutionInput;
+import graphql.ExecutionResult;
+import graphql.GraphQLError;
+import graphql.validation.ValidationError;
+
+enum DefaultGraphqlErrorHandler implements GraphqlErrorHandler {
+    INSTANCE;
+
+    @Nonnull
+    @Override
+    public HttpResponse handle(
+            ServiceRequestContext ctx,
+            ExecutionInput input,
+            ExecutionResult result,
+            @Nullable Throwable cause) {
+
+        final MediaType produceType = GraphqlUtil.produceType(ctx.request().headers());
```

Comment:
Technically, I think a user may modify the header via `ctx#updateRequest`.
Having said this, I think the chances of doing so is low, and it doesn't not make sense to modify the response type depending on the headers at this point.
SGTM